libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -202,6 +202,36 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
                                            const IotMqttCallbackInfo_t * pCallbackInfo,
                                            IotMqttOperation_t * const pOperationReference );
 
+/**
+ * @brief Sets an operation if the pointer to its address is not NULL.
+ *
+ * @param[out] pOperationReference Pointer to an operation's address.
+ * @param[in] pNewOperation Operation to set.
+ */
+static void _setOperationIfNotNull( IotMqttOperation_t * const pOperationReference,
+                                    _mqttOperation_t * pNewOperation );
+
+/**
+ * @brief Validate the arguments of @ref mqtt_function_publishasync.
+ *
+ * See @ref mqtt_function_publishasync for a description of the parameters.
+ *
+ * @return #IOT_MQTT_SUCCESS, #IOT_MQTT_NOT_INITIALIZED, or #IOT_MQTT_BAD_PARAMETER.
+ */
+static IotMqttError_t _checkPublishSetup( IotMqttConnection_t mqttConnection,
```

Comment:
Should expand _IotMqtt_ValidatePublish in iot_mqtt_validate.c, where the rest of parameter validation for publishes is performed (instead of creating a new function).

Reply:
Removed function and moved logic to _IotMatt_ValidatePublish